### PR TITLE
Issue #4981: ImportOrder with order=bottom should disallow non-static…

### PR DESCRIPTION
**Pull request by djydewang**
… import after a static import

Fix for Issue #4981

### CODE REVIEW

**Comment by romani** [comment]
@djydewang , thanks a lot.

Please provide regression report to make sure nothing unexpected will happen in real world.

**Comment by djydewang** [comment]
There is the [regression report](https://djydewang.github.io/diffReport_Issue4981/index.html).

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java [hunk 1]
```diff
@@ -401,6 +401,10 @@ else if (option == ImportOrderOption.BOTTOM) {
             }
             doVisitToken(ident, isStatic, isLastImportAndNonStatic);
 
+            if (isLastImportAndNonStatic) {
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java]: Fix was added to option `bottom`. Does `top` have a similar issue as their code looks similar?

**Comment by rnveach** [comment]
Regression looks fine to me for bottom.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java (hunk 1 above)
Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java]: @djydewang , please recheck top mode, if issue exists we need register it, if fix is complicated to do now.
I will be ok to merge even without TOP fix, as issue was for bottom only.

**Comment by djydewang** [comment]
> Does top have a similar issue as their code looks similar?

Yes!

```
bbg@king-Lenovo:~/checkstyletest$ cat configTop.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
  <property name="charset" value="UTF-8"/>

  <module name="TreeWalker">
    <module name="ImportOrder">
      <property name="groups" value="org,java"/>
      <property name="option" value="top"/>
      <property name="separated" value="true"/>
      <property name="sortStaticImportsAlphabetically" value="true"/>
    </module>
  </module>
</module>


```
```
bbg@king-Lenovo:~/checkstyletest$ cat InputImportOrderStaticGroupOrderTop_Negative.java 
package com.puppycrawl.tools.checkstyle.checks.imports.importorder;
  
import org.*;
  
import static java.lang.Math.PI;
import static org.antlr.v4.runtime.Recognizer.EOF;
  
import java.util.Set;
  
public class InputImportOrderStaticGroupOrderTop_Negative
{
 
}

```
```
bbg@king-Lenovo:~/checkstyletest$ java -jar checkstyle-8.5-all.jar -c configTop.xml InputImportOrderStaticGroupOrderTop_Negative.java 
Starting audit...
Audit done.
```
I have fixed for it.

**Comment by rnveach** [comment]
> I have fixed for it.

Please provide additional regression for it.
I will update main issue for new fix.

**Comment by djydewang** [comment]
Pitest will be OK if the Issue #5279 is fixed.

**Comment by rnveach** [comment]
This can't be merged if CI is red, and there is no point in us reviewing it more.
Why can't this be fixed in scope of this PR since this PR is what is making it red?

**Comment by djydewang** [comment]
I can fix it in this PR but I did something similar with that in Issue #5279.
So, should I fix it in this PR?

**Comment by rnveach** [comment]
If you want this PR merged first, then yes, let's just copy the fix over to here.
Once CI is green and second regression looks good, I think we will be ready to merge.

**Comment by djydewang** [comment]
It's [regression report](https://djydewang.github.io/diffReport_Issue4981_Top/index.html) for option=top.

**Comment by rnveach** [comment]
@romani Please review.

**Pull request merged**